realm/realm-annotations-processor/src/main/java/io/realm/processor/Backlink.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.processor;
+
+import java.util.List;
+
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+
+import io.realm.annotations.LinkingObjects;
+import io.realm.annotations.Required;
+
+/**
+ * A Backlink is:
+ * <p>
+ * <code>
+ * class SourceClass {
+ * // ...
+ * <p>
+ * {@literal @}LinkingObjects("targetField")
+ * RealmResults&lt;TargetClass&gt; sourceField;
+ * }
+ * </code>.
+ * <p>
+ * When managed, an instance X of a class with such a declaration will contain in the sourceField
+ * references to any instances of TargetClass whose targetField contains a reference to X.
+ * When managed, the sourceField cannot be assigned.  It can be queried normally.
+ * <p>
+ * When unmanaged, the sourceField is just another field: it can be set normally.  Managing an unmanaged
+ * instance of SourceClass destroys any previous contents of the backlinked field and replaces it with
+ * backlinks.  Unmanagning a managed instance of SourceClass nulls the backlinks field.
+ * <p>
+ * Note that, because subclassing subclasses of RealmObject is forbidden, so are constructs like:
+ * <code>RealmResults&lt;? extends Foos&lt;</code>
+ */
+final class Backlink {
+    private final VariableElement backlink;
+
+    /**
+     * The class containing the field <code>sourceField</code> with the backlink annotation
+     */
+    private final String sourceClass;
+
+    /**
+     * The name of the backlinked field, in sourceClass.
+     * The <code>RealmResults</code> field annotated with @LinkingObjects.
+     */
+    private final String sourceField;
+
+    /**
+     * The class to which the backlink, from sourceField, points
+     */
+    private final String targetClass;
```

Comment:
Now using the terms used by core: "source" is the point of the backlink, "target" is the tail.  That is, "source" is the parent, target the child.